### PR TITLE
Improve build time

### PR DESCRIPTION
**Pull request by jesrypandawa**
- enable build cache and gradle cache
- enable gradle parallel execution

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -1,11 +1,8 @@
 name: integration-test
 
 on:
-  # Trigger the workflow on push or pull request,
+  # Trigger the workflow on pull request,
   # but only for the main branch
-  push:
-    branches:
-      - main
```
Comment (prakharmathur82) [inline .github/workflows/integration.yml:8]: this we wanted on every push to main

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -11,7 +11,14 @@ jobs:
...
           java-version: 1.8
-      - name: Grant execute permission for gradlew
-        run: chmod +x gradlew
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-cache-v1-${{ hashFiles('**/*.gradle*', '**.gradle-wrapper.properties') }}
```
Comment (prakharmathur82) [inline .github/workflows/build.yml:20]: why v1? we are using cache@v2 right?

File: .github/workflows/integration.yml (hunk 1 above)
Comment (jesrypandawa) [reply .github/workflows/integration.yml:8]: do we need to run the integration test again when its pushed to main? i think running the integration test on pull request to main is enough? so we don't need to run it twice.

Comment (prakharmathur82) [reply .github/workflows/integration.yml:8]: but what about direct commits to main? (although we should discourage that but still possible)

File: .github/workflows/build.yml (hunk 2 above)
Comment (jesrypandawa) [reply .github/workflows/build.yml:20]: actually this is just the cache key name that we will sace and use. before, I'm using `gradle-cache`, but its failed to use that cache and we can not clear that. do we need to use v2 for better cache naming?

Comment (prakharmathur82) [reply .github/workflows/build.yml:20]: we can use any static word here, but it's just that v1 can be misleading

Comment (jesrypandawa) [reply .github/workflows/build.yml]: changing this to `dagger-cache`

File: .github/workflows/integration.yml (hunk 1 above)
Comment (jesrypandawa) [reply .github/workflows/integration.yml]: since there's still possibility, i'm reverting this back

**Pull request merged**